src/Interfaces/IWhitelistManagerFacet.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+/// @title Whitelist Manager Facet Interface
+/// @author LI.FI (https://li.fi)
+/// @notice Interface for WhitelistManagerFacet facet for managing approved contracts and addresses.
+/// @custom:version 1.0.0
+interface IWhitelistManagerFacet {
+    /// @notice Emitted when a new address is added to the whitelist.
+    event AddressWhitelisted(address indexed whitelistedAddress);
+
+    /// @notice Emitted when an address is removed from the whitelist.
+    event AddressRemoved(address indexed removedAddress);
+
+    /// @notice Emitted when a function signature approval is changed.
+    event FunctionSignatureApprovalChanged(
+        bytes4 indexed functionSignature,
+        bool indexed approved
+    );
+
+    /// @notice Register an address to be approved for interactions.
+    /// @param _contractAddress The contractaddress to be whitelisted.
+    function addToWhitelist(address _contractAddress) external;
+
+    /// @notice Batch register addresses to be approved for interactions.
+    /// @param _addresses The addresses to be whitelisted.
+    function batchAddToWhitelist(address[] calldata _addresses) external;
+
+    /// @notice Unregister an address from the whitelist.
+    /// @param _address The address to be removed from the whitelist.
+    function removeFromWhitelist(address _address) external;
+
+    /// @notice Batch unregister addresses from the whitelist.
+    /// @param _addresses The addresses to be removed from the whitelist.
+    function batchRemoveFromWhitelist(address[] calldata _addresses) external;
+
+    /// @notice Adds or removes a specific function signature to/from the allowlist.
+    /// @param _signature The function signature to allow or disallow.
+    /// @param _approval Whether the function signature should be allowed.
+    function setFunctionApprovalBySignature(
```

Comment:
It's a function selector, not a signature. 
Signature is "transfer(uint256)" while selector is 0x12345678

Would vouch to change this naming, too. 

Sync sigs > sync selectors

Open for other opinions as usual

Reply:
Yes that would describe it more accurately. updated!